GCEm/__init__.py

```diff
@@ -7,9 +7,201 @@
+        kernel = _get_gpflow_kernel(kernel, n_params, active_dims=active_dims, operator=kernel_op)
+    elif kernel is None:
+        # TODO Maybe this should just use the above mechanism?
+        kernel = gpflow.kernels.RBF(lengthscales=[0.5] * n_params, variance=0.01, active_dims=active_dims) + \
+                 gpflow.kernels.Linear(variance=[1.] * n_params, active_dims=active_dims) + \
+                 gpflow.kernels.Polynomial(variance=[1.] * n_params, active_dims=active_dims) + \
+                 gpflow.kernels.Bias(active_dims=active_dims)
+        print("WARNING: Using default kernel - be sure you understand the assumptions this implies")
+    else:
+        pass  # Use the user specified kernel
+
+    model = GPFlowModel(gpflow.models.GPR(data=(training_params, data.data_wrapper.data),
+                                          kernel=kernel,
+                                          noise_variance=tf.constant(noise_variance,
+                                          dtype=data.dtype)))
+
+    return Emulator(model, training_params, data, name=name, gpu=gpu)
+
+
+def _get_gpflow_kernel(names, n_params, active_dims=None, operator='add'):
+    """
+        Helper function for creating a single GPFlow kernel from a combination of kernel names.
+
+    Parameters
+    ----------
+    names: list of str
+        A list of the names of the kernels to be combined
+    n_params: int
+        The number of training parameters
+    active_dims: list of int or slice or None
+        The active dimensions to allow the kernel to fit
+    operator: {'add', 'mul'}
+        The operator to use to combine the kernels
+
+    Returns
+    -------
+    gpflow.kernels.Kernel
+        The combined GPFlow kernel
+    """
+    import gpflow.kernels
+    from operator import add, mul
+    from functools import reduce
+
+    kernel_dict = {
+        "RBF": gpflow.kernels.RBF,
+        "Linear": gpflow.kernels.Linear,
+        "Polynomial": gpflow.kernels.Polynomial,
+        "Bias": gpflow.kernels.Bias,
+        "Cosine": gpflow.kernels.Cosine,
+        "Exponential": gpflow.kernels.Exponential,
+        "Matern12": gpflow.kernels.Matern12,
+        "Matern32": gpflow.kernels.Matern32,
+        "Matern52": gpflow.kernels.Matern52,
+    }
+
+    operator_dict = {
+        'add': add,
+        'mul': mul
+    }
+
+    def init_kernel(k):
+        """
+        Initialise a GPFlow kernel with the correct shape (default) variance and lengthscales
+        """
+        if issubclass(k, gpflow.kernels.Constant):  # E.g., Bias
+            return k(active_dims=active_dims)
+        elif issubclass(k, gpflow.kernels.Linear):  # This covers polynomial too
+            return k(variance=[1.] * n_params, active_dims=active_dims)
+        elif issubclass(k, gpflow.kernels.Stationary):  # This covers e.g. RBF
+            return k(lengthscales=[1.] * n_params, active_dims=active_dims)
+        else:
+            raise ValueError("Invalid Kernel: {}".format(k))
```

Comment:
Maybe I'm being dumb here, but under what circumstances does this error ever get raised? If you put a kernel string into the 'names' input that isn't in the `kernel_dict` keys, then it fails anyways, otherwise it will pass?

Reply:
No, you're right - that's a good catch! I'll move it to cover the dictionary lookup